clang/test/CodeGenSYCL/num-simd-work-items.cpp

```diff
@@ -0,0 +1,26 @@
+// RUN: %clang_cc1 -std=c++11 -triple spir64-unknown-unknown-sycldevice -disable-llvm-passes -fsycl-is-device -emit-llvm -o - %s | FileCheck %s
+
+class Foo {
+public:
```

Comment:
Can you add a test where the number of work items is a non-type-template parameter, or otherwise dependent?  Something like:

    template<int I> class Foo2 { 
       public:
       [[intelfpga::num_simd_work_items(I)]] void operator()(){}
    };

Additionally, one where it is dependent on the template value (so take a typename T, then make it T::value).

Reply:
It's not going to work. Also it's not required to work for now. May be changed in the future.

Reply:
It seems that allowing this to just not work with an important part of the language is a giant miss.

Does it at least not crash?!

Reply:
It doesn't:
```
 num-simd-work-items.cpp:10:5: error: 'num_simd_work_items' attribute requires parameter 0 to be an integer constant
  [[intelfpga::num_simd_work_items(I)]] void operator()(){}
```

Reply:
alright then, thanks.